packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts

```diff
@@ -5,13 +5,13 @@
 
 import { catchError, expand, from, map, mergeMap, Observable, of, toArray } from "rxjs";
 import { GenericInstanceFilter, HierarchyNode, HierarchyProvider, RowsLimitExceededError } from "@itwin/presentation-hierarchies";
-import { isTreeModelHierarchyNode, TreeModelHierarchyNode, TreeModelInfoNode, TreeModelNode, TreeModelRootNode } from "./TreeModel.js";
+import { isTreeModelHierarchyNode, TreeModelError, TreeModelHierarchyNode, TreeModelRootNode } from "./TreeModel.js";
 import { createNodeId } from "./Utils.js";
 
 /** @internal */
 export interface LoadedTreePart {
   parentId: string | undefined;
-  loadedNodes: TreeModelNode[];
+  loadedNodes: (TreeModelHierarchyNode | TreeModelError)[];
```

Comment:
do we need this array to possibly contain `TreeModelError`? maybe it could return `TreeModelHierarchyNode[]`, and in case of error - just throw it?

Reply:
We would need to rewrite some portion of loader to be able to determine when it needs to make an array out of output. More importantly we want to figure out how we want to check if `loadedNodes` are an array every time we use them before the errors are merged inside the ModelNode.

Reply:
To me it looks like the loader could always return `TreeModelHierarchyNode[]`. If an error occurs, then just throw it. Then whoever calls `loadNodes` on the loader (e.g. `TreeActions`) would, instead of checking the type of returned nodes, catch the error and update the parent node.

Reply:
What about changing `LoadedTreePart` to 
```
type LoadedTreePart = {
  parentId: string | undefined;
  loadedNodes: TreeModelHierarchyNode[];
} | {
  parentId: string | undefined;
  error: TreeModelError;
}
```

Reply:
that would be okay with me

Reply:
Changed to Saulius suggestion.